Sources/Build/BuildPlan.swift

```diff
@@ -377,20 +427,18 @@ public class BuildPlan {
              }
         }
 
-        // Create product description for each product we have in the package graph.
-        self.buildProducts = graph.products().map { product in
-            // Collect all library objects.
-            var objects = product.allModules.filter{ $0.type == .library || $0.type == .test }.flatMap{ targetMap[$0]!.objects }
-
-            // Add objects from main module, if product is an executable.
-            if product.type == .executable {
-                let target = targetMap[product.executableModule]!
-                objects += target.objects
-            }
+        // Compute the products we need to build. This is all the products in the root packages and
+        // all the product dependencies of the modules we're building in the graph.
+        let products = graph.rootPackages.flatMap{$0.products} + Set(graph.modules.flatMap{$0.productDependencies})
 
-            return ProductBuildDescription(product: product, objects: objects, buildParameters: buildParameters)
+        // Create product description for each product we have in the package graph.
+        var productMap: [ResolvedProduct: ProductBuildDescription] = [:]
+        for product in products where product.type != .library(.none) {
```

Comment:
Why do we skip if product type is none here? Is that because we treat it as static and thus have nothing to do? A comment might be nice.